odoo/addons/base/models/res_users.py

```diff
@@ -656,10 +657,23 @@ def context_get(self):
         # use read() to not read other fields: this must work while modifying
         # the schema of models res.users or res.partner
         values = user.read(list(name_to_key), load=False)[0]
-        return frozendict({
+
+        context = {
             key: values[name]
             for name, key in name_to_key.items()
-        })
+        }
+
+        # ensure the language is set and is compatible with the web client
+        lang = context.get('lang') or (request and request.default_lang()) or 'en_US'
+        if lang == 'ar_AR':
+            context['lang'] = 'ar'
+        if lang in babel.core.LOCALE_ALIASES:
+            context['lang'] = babel.core.LOCALE_ALIASES[lang]
+
+        # ensure uid is set
+        context['uid'] = self.env.uid
```

Comment:
Is putting `uid` in the context still relevant?
Nowadays, `sudo()` no longer changes the uid, so `self.env.uid` or `request.env.uid` is what you look for.
(I don't expect you to fix this here, this is only a question for future cleanups.)

Reply:
The UID is still taken out of the context here and there. I tried to remove it without success, so here it is, still here.